Libraries/CLI/src/cli.c

```diff
@@ -0,0 +1,265 @@
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL MAXIM INTEGRATED BE LIABLE FOR ANY CLAIM, DAMAGES
+ * OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Except as contained in this notice, the name of Maxim Integrated
+ * Products, Inc. shall not be used except as stated in the Maxim Integrated
+ * Products, Inc. Branding Policy.
+ *
+ * The mere transfer of this software does not imply any licenses
+ * of trade secrets, proprietary technology, copyrights, patents,
+ * trademarks, maskwork rights, or any other form of intellectual
+ * property whatsoever. Maxim Integrated Products, Inc. retains all
+ * ownership rights.
+ *
+ ******************************************************************************/
+
+/* -------------------------------------------------- */
+//                      INCLUDES
+/* -------------------------------------------------- */
+
+#include "cli.h"
+
+/* -------------------------------------------------- */
+//             FUNCTION PROTOTYPES
+/* -------------------------------------------------- */
+
+/** Checks if there is white space present in the given character pointer
+ *
+ * @param p
+ * 		  character pointer containing the line accumulator input
+ *
+ * @return 1 if white space is present
+ * 		   0 if no white space is present
+ */
+bool white_space_present(char *p);
+
+/** Checks if there is no white space present in the given character pointer
+ *
+ * @param p
+ * 		  character pointer containing the line accumulator input
+ *
+ * @return 1 if no white space is present
+ * 		   0 if white space is present
+ */
+bool white_space_not_present(char *p);
+
+void User_Prompt_Sequence(void);
+
+void Console_Backspace_Sequence(void);
+
+void Console_Cmd_Clear(void);
+
+void Clear_buffer(void);
+
+/* -------------------------------------------------- */
+//                      GLOBALS
+/* -------------------------------------------------- */
+
+/*! \name buf
+    \brief Buffer to store the characters of the command 
+*/
+char buf[256];
+int idx = 0;
+
+// Define a buffer to store commands
+#define MAX_COMMAND_LENGTH 256
+
+void line_accumlator(uint8_t user_char)
+{
+    switch (user_char) {
+    case BACKSPACE: {
+        // Handle Backspace and Delete
+        if (idx > 0) {
+            //Sequence to implement a backspace on the terminal
+            Console_Backspace_Sequence();
+            idx--;
+            buf[idx] = '\0';
+        }
+        break;
+    }
+    case ENTER: {
+        // Handle Enter or carriage return
+        MXC_UART_WriteCharacter(MXC_UART_GET_UART(CONSOLE_UART), NEW_LINE);
+        MXC_UART_WriteCharacter(MXC_UART_GET_UART(CONSOLE_UART), ENTER);
+        buf[idx++] = '\r';
```

Comment:
Do we really need a carriage return in this line?
Because I see in the process command we are looking for a NULL terminated char pointer.

Reply:
The line accumulator here enters a '\r' character every time the enter key is pressed by the user.
Later the process_command function looks for carriage return while iterating over each character in the input command to check the end of the string and break from the iterating for loop.